UM/VersionUpgrade.py

```diff
@@ -19,6 +19,23 @@ def __init__(self):
         """Initialises a version upgrade plugin instance."""
 
         super().__init__()
+        self._version_regex = re.compile("\nversion ?= ?(\d+)")
+        self._setting_version_regex = re.compile("\nsetting_version ?= ?(\d+)")
+
+    def getCfgVersion(self, serialised: str) -> int:
+        """
+        Gets the version number from a config file.
+
+        In all config files that concern this version upgrade, the version number is stored in general/version, so get
+        the data from that key.
+
+        :param serialised: The contents of a config file.
+        :return: The version number of that config file.
+        """
+
+        format_version = int(self._version_regex.search(serialised).groups()[-1])
+        setting_version = int(self._setting_version_regex.search(serialised).groups()[-1])
```

Comment:
We need to have a fallback that sets the `setting_version` to 0 if it's not present?

Reply:
Isn't the profile just invalid if it's not there?

Reply:
No, profiles from before Cura 2.6 didn't have any `setting_version`.

The profile was previously marked as invalid if the normal `version` isn't there. However with your new change you also made that fall back to 0. I guess that's fine too. The profile would then get dropped rather than marked as corrupt, since there is no upgrade path from file version 0 to the most recent version.